### PR TITLE
Bootloader restart via USB command

### PR DESCRIPTION
**Pull request by stdvar**
## Description

This change will piggyback on existing HID descriptor used by shared EP for extra key and shared keyboard/mouse.
New feature report section in the HID descriptor is added to allow PC to send HID feature report to KB and thus instruct it to restart in bootloader.
Added code to distinguish between input/output/feature reports.
set_report buffer that used to handle led status is increased from (1+1 size) to (1+8 size).
This should work for any keyboard since handler uses generic bootloader_jump() function.

There is new USB_FEATURE_ENABLE option that enables shared ep just like EXTRAKEY_ENABLE. So only one option is needed but new option added for consistency in case user opted out of using EXTRAKEY or other shared ep options.

Report to send to KB is struct.pack("<II", 0x5AA555AA, 0xCC3300FF).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/protocol/chibios/usb_main.c [hunk 1]
```diff
@@ -473,6 +489,28 @@ static void    set_led_transfer_cb(USBDriver *usbp) {
     }
 }
 
+#define MAGIC0 0x5AA555AA
+#define MAGIC1 0xCC3300FF
```
Comment (dexter93) [inline tmk_core/protocol/chibios/usb_main.c]: since this isn't necessarily sonix code, maybe guard around it with a ifndef first

Comment (stdvar) [reply tmk_core/protocol/chibios/usb_main.c]: done

**Comment by tzarc** [comment]
Just a heads-up, this functionality will prevent merge to QMK upstream.

**Comment by tzarc** [comment]
See https://github.com/qmk/qmk_firmware/pull/11892 -- it allows for it to be reinstated due to downstream users relying on its functionality, but for new code this sort of addition is not going to be allowed.

**Pull request closed without merging**